engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java

```diff
@@ -276,9 +280,21 @@ private void recursivelyAddModuleDependencies(Set<Name> modules, Name moduleName
     private void addNewWorld(WorldGeneratorInfo worldGeneratorInfo) {
         selectedWorld = worldGeneratorInfo.getDisplayName() + '-' + worldNumber;
         worlds.add(new WorldSetupWrapper(new Name(worldGeneratorInfo.getDisplayName() + '-' + worldNumber), worldGeneratorInfo));
+        indexOfWorld = findIndex(worlds, selectedWorld);
         worldNumber++;
     }
 
+    /**
+     * This method refresh the dropdown menu
```

Comment:
For properly formatted Javadoc in an IDE just enter `/**<enter>` and it'll expand into a Javadoc block that includes any `return` or `param` tags etc. In this case you'll want to describe what `worldsDropdown` involves as well as indicate method-level javadoc (which is what you've got already). 

For that it tends to be better to describe the method in abstract - what is it _intended_ to be used for, not necessarily explicit places it is called from (which may change over time).

For instance: "Refreshes the worlds drop-down to catch any world name updates or a changed selection"

It may also be better to be more explicit than `refresh` - ideally get some reference to _what_ is being refreshed in the method name :-)